### PR TITLE
Replace UI and internal Giphy references

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
------------
This Pull Request brings renaming for all Giphy references inside package, classes, documentation and UI strings. Giving sequence to the [issue 11456](https://github.com/wordpress-mobile/WordPress-Android/issues/11456).

Here's the change log:

* Every UI string with `Giphy` information was replaced with `Tenor`. 

* The `giphy` packages inside `ui` and `viewmodel` were both renamed to only `gif`.

* All classes with the `Giphy` naming convention were renamed to only `Gif` to better fit the agnostic gif provider implementation strategy.

* All comments and documentation with references to `Giphy` or the Giphy SDK were readapted to better fit the `GifProvider` structure.

* Removed the Giphy SDK  reference at maven repository in `build.gradle`

Testing
------------

### Before testing ###
1. Create a Tenor API key and paste it associated with the `wp.tenor.api_key` as the value inside your `gradle.properties`
2. When testing verify that all UI strings correctly reference Tenor as the gif provider.

### Testing from the Post Editor ###
1. Click to create a new post
2. Click on the overflow button at the top right end of the screen
3. select the `Switch to classic editor` 
4. Click on the plus (+) button at the bottom left end of the screen
5. Select the `Choose from Tenor` option
6. Type in the search bar to query GIFs
7. Once inside the Gif Picker, make sure that `selection`, `preview` and `add` are all working as expected

### Testing from the Media Browser ###
1. Go to the Media Browser
2. Click on the Add button at the top right. Select the `Choose from Tenor` option
3. Type in the search bar to query GIFs
4. Once inside the Gif Picker, make sure that `selection`, `preview` and `add` are all working as expected

PR submission checklist
------------
- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -554,11 +554,11 @@ Language: de
     <string name="stats_no_data_for_period">Keine Daten für diesen Zeitraum</string>
     <string name="preference_strip_image_location">Positionsangabe von Medien entfernen</string>
     <string name="stats_cannot_be_started">Wir können die Statistiken momentan nicht öffnen. Bitte versuche es später erneut</string>
-    <string name="giphy_powered_by_giphy">Bereitgestellt von GIPHY</string>
```
Comment (planarvoid) [inline WordPress/src/main/res/values-de/strings.xml:557]: Maybe I didn't mention this explicitly. We shouldn't change any of the translation files. It should all go through the translation service. You should just change the names of the string items in the basic strings.xml (and you can also delete the translations of the original strings). I think (I'm not sure here) they'll get overriden by the service result anyway.

**Review by planarvoid — COMMENTED** [review]
Overall looks pretty great! I have one comment about the translations but that should be fairly simple to change. Thanks for all the changes 👍

File: WordPress/src/main/res/values-de/strings.xml (hunk 1 above)
Comment (ThomazFB) [reply WordPress/src/main/res/values-de/strings.xml:557]: Just to be sure before I remove those changes, should I also revert the translated strings which only got renamed? Or this revert applies only to the strings in which the content was changed?

Comment (ThomazFB) [reply WordPress/src/main/res/values-de/strings.xml:557]: Thanks for the feedback! I just deleted all translation strings as suggested and left the renaming and the Tenor replacement only at the original `strings.xml` file, would be fine that way? Here's the commit: 8eabca9ae83f3936c3ae438aca70e22e3908dca5 . Thanks!

**Review by planarvoid — APPROVED** [review]
Thanks for the changes! It all looks great :-). I'll leave the final round of stress-testing for the final PR 👍

**Pull request merged**